src/utils.jl

```diff
@@ -77,6 +77,44 @@ function sideof(p::Point{2,T}, c::Chain{2,T}) where {T}
   ifelse(isapprox(w, zero(T), atol=atol(T)), :OUTSIDE, :INSIDE)
 end
 
+"""
+    sideof(point, mesh)
+Determines whether a `point` is inside, outside the surface of a `mesh`. returns :INSIDE, :OUTSIDE, or :ON. Uses a ray-casting algorithm.
+"""
+function sideof(point::Point{3,T}, mesh::Mesh{3,T}) where {T}
+  if !(eltype(mesh) <: Triangle)
+    throw(ArgumentError(
+          "This function only works for surface meshes with triangles as elements."))
+  end
+  z = last.(coordinates.(extrema(mesh)))
+  r = Ray(point, Vec(zero(T), zero(T), 2 * (z[2] - z[1])))
+  vs = vertices(mesh)
+  touching_types = (EdgeTouchingRayTriangle, CornerTouchingRayTriangle, TouchingRayTriangle)
+
+  intersects = false
+  edge_crosses = 0
+  corners_crossing = Point{3,T}[]
+  for elem in mesh
+    I = intersection(r, elem)
+    if type(I) == CrossingRayTriangle
+      intersects = !intersects
+    elseif type(I) ∈ touching_types
```

Comment:
```suggestion
    elseif type(I) ∈ [EdgeTouchingRayTriangle, CornerTouchingRayTriangle, TouchingRayTriangle]
```

Reply:
I think we should we use a tuple - () instead of array - [] here so the compiler knows it is a fixed length. Perhaps it doesn't make a difference in this specific case, but I think its better to use an immutable container when you can.